tools/projmgr/docs/Manual/Overview.md

```diff
@@ -0,0 +1,1317 @@
+
+Let’s assume a configuration file named “MyConfigFile.h” with version 1.2.0 which is part of component “MyComponent” and the destination directory is RTE/MyComponentClass/
+
+1. “MyComponent” is selected by the project/layer
+if `RTE/MyComponentClass/MyConfigFile.h` does not exist `MyConfigFile.h` will be copied from the pack where the component version of MyComponent is present `RTE/MyComponentClass/MyConfigFile.h` in addition a copy of this file will be created `RTE/MyComponentClass/.MyConfigFile.h-1.2.0`. The user will only see and edit the `MyConfigFile.h`.
+
+2. If a new version of the component with a new version `1.3.0` of the config file becomes available the tools are able to:
+    - read from the filename `.MyConfigFile.h-1.2.0` that the current file is based on version 1.2.0 
+    - that the new configfile `1.3.0` is greater in the minimum version only -> new feature
+    - if the tools run a simple diff on `MyConfigFile.h` and `.MyConfigFile.h-1.2.0` “local” modifications can be identified with
+         - (a) print a user notification about the config file not being up-to-date
+         - (b) run a 3-way merge utility attempting to merge the files if it fails revert back.
+
+
+## Output versions to *.cprj
+
+The ProjMgr should always generate *.cprj files that contain version information about the used software packs.
+
+## CMSIS-Zone Integration
+
+Add to the project the possiblity to specify.  The issue might be that the project files become overwelming, alternative is to keep partitioning in separate files.
+
+```yml
+memories:              # specifies the required memory
+  - split: SRAM_NS
+    into:
+      - region: DATA_NS
+        size: 128k
+        permission: n
+    
+peripherals:           # specifies the requried peripherals
+  - peripheral: I2C0
+    permission: rw, s
+```
+
+## Layer Interface Defintions
+
+A software layer could specify the interfaces that it provides.  The interface specification indicates also the configuration of the layer.  Issue might be that a standardization across the industry is required.
+
+```yml
+interfaces:
+  consumes:
+    - RTOS2:     # requires CMSIS-RTOS2 features
+  provides:
+    - C_VIO:     # provides CMSIS-VIO interface for LEDs
+    - A_IO9_I:   # provides Audrino connector pin I09 configured as input
+    - A_IO10_O:  # provides Audrino connector pin I10 configured as output
+    - STDOUT:    # redirects STDIO
+    - STDIN:
+    - STDERR:
+    - Heap: 65536 # provides 64K heap
+```
+
+## CMSIS-Pack extensions
+
+### Board condition
+
+It should be possible to use conditions to filter against a selected board name.
+
+### Layers in packs
+
+A layer is a set of pre-configured software components. It should be possible to store a layer in a pack and apply filter conditions to it.  In combination with interfaces specifications, an interactive IDE should be able to display suitable layers that could be added to an application.
```

Comment:
The interactive UI is indeed the point...
All these yml files are not easy to maintain manually...despite the YML format.

And who should make the checks between the pdsc conditions and the YML content ?

Reply:
Agreed.  We need IDE flows for it.  We will explain this afternoon